### ISSUE TITLE
FastaConverter.FastaDescriptionLine not kryo-registered

### ISSUE THREAD

**Issue by ryan-williams** [bigdatagenomics/adam#893]
Running some apps with `spark.kryo.registrationRequired=true` and getting failures due to this.

Pretty sure the fix is just:

```
kryo.register(classOf[FastaConverter.FastaDescriptionLine])
```

in `ADAMKryoRegistrator`

**Comment by ryan-williams** [comment]
Also `ReferenceRegion`, which @ipurusho is hitting.

**Comment by fnothaft** [comment]
@ryan-williams did you have a fix for this/code that reproduces the issue? I'm a bit surprised about any issues with `ReferenceRegion`, can you elaborate?

**Comment by ryan-williams** [comment]
I was aggressively auditing my serialization story and running with `spark.kryo.registrationRequired=true`, and it was pretty trivial to uncover these. I'll think about exactly what would do it, but any join on a `ReferenceRegion` should do it, I'd think.. anything with a shuffle stage involving records of the types in question that get ser'd...

**Comment by fnothaft** [comment]
Ah, great! I missed that you were running with `spark.kryo.registrationRequired=true`. This makes perfect sense now; sorry for the confusion.

**Comment by fnothaft** [comment]
Resolved by #1020.
